### PR TITLE
1 - API Files for submitting and cancelling jobs

### PR DESCRIPTION
**Pull request by mohsenbakhit**
# ADD API files for submitting and cancelling jobs

## Added functionality for submitting files to SLURM using sbatch and cancelling with scancel.

- **Issue:** #1 
- **Type:** Feature

## Changes

Describe the changes made in this PR:

1. **Change 1:**  Refactored submit_job.py with accordance to the function calls in main
2. **Change 2:** Refactored cancel_job.py with accordance to the function calls in main
3. **Change 3:** Moved clean_up and get_slurm_id to util.py which is where common util functions will lie.

## Checklist

Please ensure the following tasks are complete:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ubchemica — APPROVED** [review]
Hi Mohsen,

Just a couple of comments...
cancel.py shouldn't do a cleanup at the end because useful intermediate results may have been generated.  is there some particular reason that you don't let main.py get the SLURM ID and then have this as the argument for cancel?  In principle, this function should return a status depending upon what scancel does.  For example, imagine the job has already finished, or died on its own.... I think this would cause scancel to return an error, or at least some sort of information.

In all these scripts, I noticed you are referencing path to '.' .  The main top-level directory in which all the job subdirectories are created should be passed as a variable (that could be hard-wired in main.py, for example).

**Comment by mohsenbakhit** [comment]
@ubchemica I have removed the auto-cleanup from cancel and have changed the top-level directory to be a variable in main that modules can import. 
The reason I don't like main.py getting involved in getting the slurm_id is mainly from the perspective of the code feels cleaner to me if the tasks handle that part itself. If your preference is for it to be in main, I can do that. 
I have also made both cancel and submit return the status in stdout and stderr for the backend to understand the status of the job.

**Review by ubchemica — APPROVED** [review]

**Review by lykawang — APPROVED** [review]
LGTM!

File: main.py [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
\ No newline at end of file
+    raw_json = input()
+    try:
+        job_input_data = json.loads(raw_json)
+        action = job_input_data["action"]
+    except:
+        print("Error parsing JSON")
+    match action:
+        case "submit":
+            submit_job(job_input_data)
```
[the remark this answers is not in the document]
Comment (mohsenbakhit) [reply main.py]: Adding a variable would be a more sophisticated approach.

File: util.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    Args: a string with the id given to the job by the db
+
+    Returns: a string with the id given to the job by slurm
+    """
+    current_path = os.path.join("./"+db_job_id)
+    slurm_job_id = ""
+    try:
+        with open(current_path + "/slurm_id.txt", "r") as file:
+            slurm_job_id = file.readLine()
```
[the remark this answers is not in the document]
Comment (mohsenbakhit) [reply util.py]: Damn, slipped my eyes. Amended.

**Pull request merged**